### DISCUSSION TITLE
How to reduce duplicate names in the display?

### DISCUSSION THREAD

**Discussion by nanpeng** [organicmaps/organicmaps#4583] [Q&A]
Due to poor  mwm data quality, the same road may be divided into too many sub-roads, each with the same name set. Therefore, there are too many duplicate names displayed. Currently, it is not easy to integrate the data. Is there any way to minimize the display of duplicate names as much as possible?

**Comment by patepelo** [comment]
> Due to poor  mwm data quality, the same road may be divided into too many sub-roads, each with the same name set. Therefore, there are too many duplicate names displayed. Currently, it is not easy to integrate the data. Is there any way to minimize the display of duplicate names as much as possible?

Maybe the OSM streets, at least the complex, need to be joint by relations, the way rivers are.
